packages/i18n-calypso/src/use-translate.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+export default function( i18n ) {
+	const translate = i18n.translate.bind( i18n );
+	const getLocaleSlug = i18n.getLocaleSlug.bind( i18n );
+
+	return function useTranslate() {
+		const [ localeSlug, setLocaleSlug ] = React.useState( getLocaleSlug );
+
+		React.useEffect(() => {
+			const onChange = () => setLocaleSlug( getLocaleSlug() );
```

Comment:
I ran into some confusing issues in my own PR when dealing with subscriptions for multiple instances of the same class. This code looks good, but I'd suggest testing reactivity when you have multiple instances of the same component, and ensure they all get updated correctly.